public/app/core/reducers/appNotification.ts

```diff
@@ -16,30 +22,107 @@ const appNotificationsSlice = createSlice({
   initialState,
   reducers: {
     notifyApp: (state, { payload: newAlert }: PayloadAction<AppNotification>) => {
-      if (Object.values(state.byId).some((alert) => isSimilar(newAlert, alert))) {
+      if (Object.values(state.byId).some((alert) => isSimilar(newAlert, alert) && alert.showing)) {
         return;
       }
 
       state.byId[newAlert.id] = newAlert;
+      serializeNotifications(state.byId);
```

Comment:
Having `serializeNotifications` called manually for each reducer isn't great IMO, and there are "better" ways like subscribing to the redux store and serializing then, but I wanted to avoid serializing for _every_ dispatched action.
Could maybe make a wrapper action like `doXAndSerialize` that accepts an action and then serializes, maybe?